service/frontend/workflow_handler.go

```diff
@@ -3829,27 +3841,66 @@ func (wh *WorkflowHandler) validateBuildIdCompatibilityUpdate(
 	return nil
 }
 
-func (wh *WorkflowHandler) historyArchived(ctx context.Context, request *workflowservice.GetWorkflowExecutionHistoryRequest, namespaceID namespace.ID) bool {
-	if request.GetExecution() == nil || request.GetExecution().GetRunId() == "" {
+func (wh *WorkflowHandler) historyArchived(ctx context.Context, execution *commonpb.WorkflowExecution, namespaceID namespace.ID) bool {
```

Comment:
Any particular reason to change the func argument type to execution?

Reply:
I would prefer to keep the original func signature, i.e. history request as argument, to be consistent with the func name and reserve for future extension. 
```
func (wh *WorkflowHandler) historyArchived(ctx context.Context, request *workflowservice.GetWorkflowExecutionHistoryRequest, namespaceID namespace.ID) bool {
	return wh.workflowExecutionArchived(ctx, request.execution, namespaceID)
}
```